src/Analyzers/CSharp/Analyzers/UseCollectionExpression/CSharpUseCollectionExpressionForArrayDiagnosticAnalyzer.cs

```diff
@@ -6,68 +6,30 @@
-        new LocalizableResourceString(nameof(AnalyzersResources.Collection_initialization_can_be_simplified), AnalyzersResources.ResourceManager, typeof(AnalyzersResources)),
-        isUnnecessary: false);
-
-    private static readonly DiagnosticDescriptor s_unnecessaryCodeDescriptor = CreateDescriptorWithId(
-        IDEDiagnosticIds.UseCollectionExpressionForArrayDiagnosticId,
-        EnforceOnBuildValues.UseCollectionExpressionForArray,
-        new LocalizableResourceString(nameof(AnalyzersResources.Simplify_collection_initialization), AnalyzersResources.ResourceManager, typeof(AnalyzersResources)),
-        new LocalizableResourceString(nameof(AnalyzersResources.Collection_initialization_can_be_simplified), AnalyzersResources.ResourceManager, typeof(AnalyzersResources)),
-        isUnnecessary: true);
-
     public CSharpUseCollectionExpressionForArrayDiagnosticAnalyzer()
-        : base(ImmutableDictionary<DiagnosticDescriptor, IOption2>.Empty
-                .Add(s_descriptor, CodeStyleOptions2.PreferCollectionExpression)
-                .Add(s_unnecessaryCodeDescriptor, CodeStyleOptions2.PreferCollectionExpression))
+        : base(IDEDiagnosticIds.UseCollectionExpressionForArrayDiagnosticId,
+               EnforceOnBuildValues.UseCollectionExpressionForArray)
     {
     }
 
-    protected override void InitializeWorker(AnalysisContext context)
-        => context.RegisterCompilationStartAction(context =>
-        {
-            if (!context.Compilation.LanguageVersion().SupportsCollectionExpressions())
-                return;
-
-            // We wrap the SyntaxNodeAction within a CodeBlockStartAction, which allows us to
-            // get callbacks for object creation expression nodes, but analyze nodes across the entire code block
-            // and eventually report fading diagnostics with location outside this node.
-            // Without the containing CodeBlockStartAction, our reported diagnostic would be classified
-            // as a non-local diagnostic and would not participate in lightbulb for computing code fixes.
-            context.RegisterCodeBlockStartAction<SyntaxKind>(context =>
-            {
-                context.RegisterSyntaxNodeAction(
-                    context => AnalyzeArrayInitializerExpression(context),
-                    SyntaxKind.ArrayInitializerExpression);
-
-                context.RegisterSyntaxNodeAction(
-                    context => AnalyzeArrayCreationExpression(context),
-                    SyntaxKind.ArrayCreationExpression);
-            });
-        });
-
-    private static void AnalyzeArrayCreationExpression(SyntaxNodeAnalysisContext context)
+    protected override void InitializeWorker(CodeBlockStartAnalysisContext<SyntaxKind> context)
+    {
+        context.RegisterSyntaxNodeAction(AnalyzeArrayInitializerExpression, SyntaxKind.ArrayInitializerExpression);
+        context.RegisterSyntaxNodeAction(AnalyzeArrayCreationExpression, SyntaxKind.ArrayCreationExpression);
```

Comment:
subtypes only ned to do this registration.  all the outside parts are handled by the base type.